agent/lib/kontena/observable.rb

```diff
@@ -0,0 +1,72 @@
+module Kontena
+  # An Actor that has some value
+  # The value does not yet exist when initialized, it is nil
+  # Once the value is first updated, then other Actors will be able to observe it
+  # When the value later updated, other Actors will also observe those changes
+  module Observable
+    include Kontena::Logging
+
+    def value
+      @value
+    end
+
+    # Registered Observers
+    #
+    # @return [Hash{Observe => Celluloid::Proxy::Cell<Observer>}]
+    def observers
+      @observers ||= {}
+    end
+
+    # The Observable has a value.
+    #
+    # This will notify any Observers, causing them to yield if ready.
+    #
+    # @param value [Object]
+    # @raise [ArgumentError] Update with nil value
+    def update(value)
```

Comment:
`update_observable(...)` and `reset_observable`? It's a little verbose, but not too much.